### PR TITLE
[CWS] Optimises the async flag retrieving 

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

Reduce the payload size needed for retrieving the async flag for every event by 8octs.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/ebpf/c/defs.h [hunk 1]
```diff
@@ -239,8 +239,9 @@ struct kevent_t {
 };
 
 struct syscall_t {
-    s64 retval;
-    u64 async; /* TODO: optimize the way we retrieve the async boolean */
+    s64 retval:32,
```
Comment (lebauce) [inline pkg/security/ebpf/c/defs.h:242]: What about using `kevent_t.type` instead ? If syscalls returns a 64 bits integers, I would prefer not to shrink the value returned by the kernel

Comment (spikat) [reply pkg/security/ebpf/c/defs.h]: You're right, I just updated it, thanks.

**Review by cswatt — APPROVED** [review]

File: pkg/security/ebpf/c/link.h [hunk 2]
```diff
@@ -201,7 +201,7 @@ int __attribute__((always_inline)) dr_link_dst_callback(void *ctx, int retval) {
         .event.type = EVENT_LINK,
         .event.timestamp = bpf_ktime_get_ns(),
         .syscall.retval = retval,
-        .syscall.async = (u64)syscall->async,
+        .event.async = (u64)syscall->async,
```
Comment (paulcacheux) [inline pkg/security/ebpf/c/link.h:204]:
```suggestion
        .event.async = syscall->async,
```

File: pkg/security/secl/model/unmarshallers.go [hunk 3]
```diff
@@ -72,7 +72,13 @@ func (e *Event) UnmarshalBinary(data []byte) (int, error) {
 	}
 
 	e.TimestampRaw = ByteOrder.Uint64(data[8:16])
-	e.Type = ByteOrder.Uint64(data[16:24])
+	e.Type = ByteOrder.Uint32(data[16:20])
+	if data[20] != 0 {
```
Comment (paulcacheux) [inline pkg/security/secl/model/unmarshallers.go:76]: Are we sure the bitfield will actually be at the start ? (Like in a big-endian situation)

File: pkg/security/ebpf/c/link.h (hunk 2 above)
Comment (spikat) [reply pkg/security/ebpf/c/link.h:204]: Fixed, thanks

File: pkg/security/secl/model/unmarshallers.go (hunk 3 above)
Comment (spikat) [reply pkg/security/secl/model/unmarshallers.go:76]: Yes you're right, I removed the bitfield. Thanks

**Review by paulcacheux — APPROVED** [review]

**Pull request merged**